scripts/memory_logger.py

```diff
@@ -1,110 +1,149 @@
-    done = False
-    while not done:
-      if self.listener.WaitForEvent(lldb.UINT32_MAX, event):
-        state = lldb.SBProcess.GetStateFromEvent(event)
-        if state == lldb.eStateExited:
-          done = True
-        elif state == lldb.eStateStopped:
-          lldb_results = self._run_commands(['process status', 'bt', 'cont'])
-          done = True
-        elif state == lldb.eStateRunning:
-          self._run_commands(['process interrupt'])
-      if state == lldb.eStateCrashed or state == lldb.eStateInvalid or state == lldb.eStateExited:
-        return 'Binary exited before sample could be taken'
-      time.sleep(0.03)
-
-    # Due to some strange race condition we have to wait until eState is running
-    # before we can pass the 'detach, quit' command. Why we can not do this all in
-    # one go... bug?
-    done = False
-    while not done:
-      if self.listener.WaitForEvent(lldb.UINT32_MAX, event):
-        state = lldb.SBProcess.GetStateFromEvent(event)
-        if state == lldb.eStateRunning:
-          self._run_commands(['detach', 'quit'])
-          done = True
-      if state == lldb.eStateCrashed or state == lldb.eStateInvalid or state == lldb.eStateExited:
-        return 'Binary exited before sample could be taken'
-      time.sleep(0.03)
-    return self._parseStackTrace(lldb_results)
-
-class GDB:
-  def _parseStackTrace(self, gibberish):
-    not_gibberish = re.findall(r'\(gdb\) (#.*)\(gdb\)', gibberish, re.DOTALL)
+    not_gibberish = re.findall(r'\(' + self.debugger + '\) (#.*)\(' + self.debugger + '\)', gibberish, re.DOTALL)
     if len(not_gibberish) != 0:
       return not_gibberish[0]
     else:
-      return 'Stack Trace failed:', gibberish
-
-  def _waitForResponse(self, wait=True):
-    while wait:
-      self.gdb_stdout.seek(self.last_position)
-      for line in self.gdb_stdout:
-        if line == '(gdb) ':
-          self.last_position = self.gdb_stdout.tell()
-          return True
-      time.sleep(0.05)
-    time.sleep(0.05)
-    return True
+      # return a blank line, as to not polute the log. gibberish here usually indicates a bunch of warnings or information about loading symbols
+      return ''
 
-  def getStackTrace(self, pid):
-    gdb_commands = [ 'attach ' + pid + '\n', 'set verbose off\n', 'thread\n', 'apply\n', 'all\n', 'bt\n', 'quit\n', 'y\n' ]
-    self.gdb_stdout = SpooledTemporaryFile()
-    self.last_position = 0
-    gdb_process = subprocess.Popen([which('gdb'), '-nx'], stdin=subprocess.PIPE, stdout=self.gdb_stdout, stderr=self.gdb_stdout)
-    while gdb_process.poll() == None:
-      for command in gdb_commands:
-        if command == gdb_commands[-1]:
-          gdb_commands = []
-        elif self._waitForResponse():
-          # I have seen GDB exit out from under us
+  def _waitForResponse(self, dbg_stdout):
+    # allow a maximum of 5 seconds to obtain a debugger prompt position. Otherwise we can hang indefinitely
+    end_queue = time.time() + float(10)
```

Comment:
I need to update either the comment or modify this back to 5 seconds. I had it set to 10 while I was testing my changes.